File: CS/Server/BallGameServer/ServerProgram.cs
using System;
using System.Collections.Generic;
using System.Net;
using System.Net.Sockets;
using System.Threading;

namespace BallGameServer
{
    class ServerProgram
    {
        private const int PORT = 8888;
        public static List<ClientHandler> clients = new List<ClientHandler>();
        private static void RunServer()
        {
            TcpListener listener = new TcpListener(IPAddress.Loopback, PORT);
            listener.Start();
            Console.WriteLine("Waiting for connections...");
            while (true)
            {
                TcpClient tcpClient = listener.AcceptTcpClient();
                ClientHandler newClient = new ClientHandler(tcpClient);
                new Thread(newClient.Run).Start();
                clients.Add(newClient);
            }
        }

        static void Main(string[] args)
        {
            RunServer();
        }
    }
}

File: Java/Server/src/ServerProgram.java
import java.net.ServerSocket;
import java.net.Socket;
import java.util.ArrayList;
import java.util.List;

public class ServerProgram {

    private static final int PORT = 8888;
    static List<ClientHandler> clients = new ArrayList<>();

    @SuppressWarnings("InfiniteLoopStatement")
    private static void runServer() {
        ServerSocket serverSocket;
        try {
            serverSocket = new ServerSocket(PORT);
            System.out.println("Waiting for connections...");
            while (true) {
                Socket socket = serverSocket.accept();
                ClientHandler newClient = new ClientHandler(socket);
                new Thread(newClient).start();
                clients.add(newClient);
            }
        } catch (Exception e) {
            System.err.println(e.getMessage());
        }
    }

    public static void main(String[] args) {
        runServer();
    }
}
